### PR TITLE
Update/2303

### PR DESCRIPTION
**Pull request by jrconlin**
## Description

Update libraries and calls for 2023 (add comments, required features, etc.)

## Testing

Well, there's one test at least.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,10 @@
 # Docker 17.05 or higher required for multi-stage builds
-FROM rust:1.48 as builder
+FROM rust:1.68 as builder
```
Comment (pjenvey) [inline Dockerfile]: Apparently `rust:1.68` is now based on `bullseye`. Let's specify the same debian release in both the builder and the final image so they don't go out of sync.

```suggestion
FROM rust:1.68-bullseye as builder
```
(or `buster`)

File: Dockerfile [hunk 2]
```diff
@@ -17,7 +16,7 @@ RUN \
     cp /app/target/release/${APPNAME} /app/bin
 
 
-FROM debian:stretch-slim
+FROM debian:buster-slim
```
Comment (pjenvey) [inline Dockerfile]: (or keep `buster`)
```suggestion
FROM debian:bullseye-slim
```

File: src/server/mod.rs [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
-use crate::{
-    error::HandlerError,
-    metrics,
-    settings::Settings,
-    web::{handlers, middleware},
-};
+use crate::web::middleware::sentry::SentryWrapper;
+use crate::{error::HandlerError, metrics, settings::Settings};
+
+mod dockerflow;
```
Comment (pjenvey) [inline src/server/mod.rs:12]: Forgot to git add `dockerflow.rs`

File: src/server/mod.rs [hunk 4]
```diff
@@ -44,37 +44,26 @@ macro_rules! build_app {
...
+            .service(web::resource("/__heartbeat__").route(web::get().to(dockerflow::health_route)))
             .service(
-                web::resource("/__version__").route(web::get().to(|_: HttpRequest| {
-                    // return the contents of the version.json file created by circleci
-                    // and stored in the docker root
-                    HttpResponse::Ok()
-                        .content_type("application/json")
-                        .body(include_str!("../../version.json"))
-                })),
+                web::resource("/__lbheartbeat__")
```
Comment (pjenvey) [inline src/server/mod.rs]: Let's use `configure` for all the dockerflow endpoints here (see [merino for an example](https://github.com/mozilla-services/merino/blob/main/merino-web/src/lib.rs#L140))

File: src/server/mod.rs [hunk 5]
```diff
@@ -44,37 +44,26 @@ macro_rules! build_app {
...
+                    .route(web::get().to(dockerflow::lb_heartbeat_route)),
             )
-            .service(web::resource("/__error__").route(web::get().to(handlers::test_error)))
+            .service(web::resource("/__status__").route(web::get().to(dockerflow::status_route)))
```
Comment (pjenvey) [inline src/server/mod.rs]: Should probably kill the non dockerflow `__status__` as we already have `__heartbeat__` and `__version__`, doesn't it come from autopush's specific `/status`?

**Review by pjenvey — DISMISSED** [review]

File: Dockerfile (hunk 1 above)
Comment (jrconlin) [reply Dockerfile]: Going with Buster for now.

File: src/web/mod.rs [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 pub mod extractors;
-pub mod handlers;
 pub mod middleware;
 
 // Known DockerFlow commands for Ops callbacks
-pub const DOCKER_FLOW_ENDPOINTS: [&str; 4] = [
+pub const DOCKER_FLOW_ENDPOINTS: [&str; 5] = [
     "/__heartbeat__",
     "/__lbheartbeat__",
+    "/__status__",
```
Comment (pjenvey) [inline src/web/mod.rs]:
```suggestion
```

**Review by pjenvey — CHANGES_REQUESTED** [review]

**Review by pjenvey — APPROVED** [review]

**Pull request merged**